executable/build.go

```diff
@@ -39,7 +39,13 @@ type Build struct {
 func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 	result := libcnb.NewBuildResult()
 
-	execJar, err := LoadExecutableJAR(context.Application.Path)
+	cr, err := libpak.NewConfigurationResolver(context.Buildpack, nil)
```

Comment:
There is a hidden side-effect in NewConfigurationResolver. When you run it *if* there is a logger passed into it, then it will log the configuration state. This is handy for users because it lets them see the values that the buildpack sees and what it's using.

We were not logging this before because there was no configuration, but I think we should log it now that we're introducing this new env variable.

The problem is that you cannot log that information until after you've determined that the buildpack is going to run and after we log the title.

Because NewConfigurationResolver does this as a side effect, what you essentially need to do is to recreate the object. So line 42 is OK, but down below the title where the NewConfigurationResolver used to be, we want to reinitialize NewConfigurationResolver and pass in the logger. That will let you use the config resolver and get the output we want for the user.

Reply:
👍 - I added a note below.